adapters/repos/db/shard.go

```diff
@@ -73,7 +92,13 @@ func NewShard(ctx context.Context, promMetrics *monitoring.PrometheusMetrics,
 	}
 
 	invertedIndexConfig := index.getInvertedIndexConfig()
-
+	val, ok := os.LookupEnv("MaxNumberGoroutinesFactor")
+	var MaxNumberGoroutinesFactor float64
+	if !ok {
+		MaxNumberGoroutinesFactor = 2
+	} else {
+		MaxNumberGoroutinesFactor, _ = strconv.ParseFloat(val, 32)
```

Comment:
What is the expected behavior here if the parse fails, due to an invalid value set in the env var? I think in this case, the returned `val` would just be 0. 

Maybe don't ignore the error, and if one occurs, log a server message saying that the value was unable to be parsed, and fall back to the default value?

Reply:
I wasn't sure if this environment variable is something that is useful? It doesn't seem to make that big of a difference speed-wise on my machine, but I haven't measured the memory difference. If it should stay in, I'll add tests :)

Reply:
As discussed in the standup this stays in. I added a test, let me know what you think